pkg/ingester/errors_test.go

```diff
@@ -348,151 +348,305 @@ func TestWrapOrAnnotateWithUser(t *testing.T) {
+	timestamp := model.Time(1)
 
+	testCases := map[string]struct {
+		err                 error
+		doNotLogExpected    bool
+		expectedCode        codes.Code
+		expectedMessage     string
+		expectedDetails     *mimirpb.WriteErrorDetails
+		expectedTranslation error
+	}{
+		"a generic error gets translated into an Internal gRPC error without details": {
+			err:             originalErr,
+			expectedCode:    codes.Internal,
+			expectedMessage: originalMsg,
+			expectedDetails: nil,
+		},
+		"a DoNotLog of a generic error gets translated into an Internal gRPC error without details": {
+			err:              log.DoNotLogError{Err: originalErr},
+			expectedCode:     codes.Internal,
+			expectedMessage:  originalMsg,
+			expectedDetails:  nil,
+			doNotLogExpected: true,
+		},
+		"an unavailableError gets translated into an errorWithStatus Unavailable error with details": {
+			err:             newUnavailableError(services.Stopping),
+			expectedCode:    codes.Unavailable,
+			expectedMessage: newUnavailableError(services.Stopping).Error(),
+			expectedDetails: &mimirpb.WriteErrorDetails{Cause: mimirpb.SERVICE_UNAVAILABLE},
+		},
+		"a wrapped unavailableError gets translated into a non-loggable errorWithStatus Unavailable error": {
+			err:             fmt.Errorf("wrapped: %w", newUnavailableError(services.Stopping)),
+			expectedCode:    codes.Unavailable,
+			expectedMessage: fmt.Sprintf("wrapped: %s", newUnavailableError(services.Stopping).Error()),
+			expectedDetails: &mimirpb.WriteErrorDetails{Cause: mimirpb.SERVICE_UNAVAILABLE},
+		},
+		"an instanceLimitReachedError gets translated into a non-loggable errorWithStatus Unavailable error with details": {
+			err:              newInstanceLimitReachedError("instance limit reached"),
+			expectedCode:     codes.Unavailable,
+			expectedMessage:  newInstanceLimitReachedError("instance limit reached").Error(),
+			expectedDetails:  &mimirpb.WriteErrorDetails{Cause: mimirpb.INSTANCE_LIMIT},
+			doNotLogExpected: true,
+		},
+		"a wrapped instanceLimitReachedError gets translated into a non-loggable errorWithStatus Unavailable error with details": {
+			err:              fmt.Errorf("wrapped: %w", newInstanceLimitReachedError("instance limit reached")),
+			expectedCode:     codes.Unavailable,
+			expectedMessage:  fmt.Sprintf("wrapped: %s", newInstanceLimitReachedError("instance limit reached").Error()),
+			expectedDetails:  &mimirpb.WriteErrorDetails{Cause: mimirpb.INSTANCE_LIMIT},
+			doNotLogExpected: true,
+		},
+		"a tsdbUnavailableError gets translated into an errorWithStatus Internal error with details": {
+			err:             newTSDBUnavailableError("tsdb stopping"),
+			expectedCode:    codes.Internal,
+			expectedMessage: newTSDBUnavailableError("tsdb stopping").Error(),
+			expectedDetails: &mimirpb.WriteErrorDetails{Cause: mimirpb.TSDB_UNAVAILABLE},
+		},
+		"a wrapped instanceLimitReachedError gets translated into a non-loggable errorWithStatus Internal error with details": {
```

Comment:
if this error is non-loggable shouldn't it have `doNotLogExpected: true,`?

Reply:
This was just a mistake in the test description: it should be `a wrapped tsdbUnavailableError gets translated...`, and this error is loggable. The actual `instanceLimitReachedError` tests (several lines above) is a non-loggable test, and in those tests `doNotLogExpected` is set to `true`.

I have corrected the test description.